### PR TITLE
[#ICC-232] Add MessageStatus ChangeFeed for Reminder

### PR DESCRIPTION
**Pull request by AleDore**
#### List of Changes

- Add `CosmosApiMessageStatusChangeFeedForReminder` handler
- Add `messageStatusAvroFormatter`
- Add `message-status` avro schema
- Add tests

#### Motivation and Context

This handler is required by Reminder in order to know status change on citizen's messages, such as readStatus.
#### How Has This Been Tested?

Unit Test
#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by AleDore** [comment]
Infra [PR](https://github.com/pagopa/io-infra/pull/233)

**Review by balanza — APPROVED** [review]
See no evil

File: CosmosApiMessageStatusChangeFeedForReminder/handler.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  // eslint-disable-next-line functional/immutable-data
+  context.bindings.outputMessageStatus = pipe(
+    rawMessageStatus,
+    RA.map(RetrievedMessageStatus.decode),
+    RA.rights,
+    RA.filter(
+      messageStatus => messageStatus.status === MessageStatusValueEnum.PROCESSED
+    ),
+    RA.map(toAvroMessageStatus)
+  );
```
Comment (balanza) [inline CosmosApiMessageStatusChangeFeedForReminder/handler.ts:21]: Maybe is safer to only assign value if message array contains at least 1 element?

Comment (AleDore) [reply CosmosApiMessageStatusChangeFeedForReminder/handler.ts]: Done ;)

**Review by gquadrati — APPROVED** [review]
LGTM

**Pull request merged**